File: RacingLeagueHub/BLL/Entities/Resource.cs
namespace RacingLeagueHub.Entities;

public class Resource : EntityBase
{
    public string Url { get; set; }
    public string FileName { get; set; }
    public string Extension { get; set; }
    public string MimeType { get; set; }
    public long SizeInBytes { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public bool? IsThumbnail { get; set; }
}

File: RacingLeagueHubClient/src/app/shared/models/resource.ts
import { BaseDto } from "./dtos";

export interface ResourceDto extends BaseDto {
    storageId: string;
    fileUrl: string;
    extension: string;
    fileName: string;
    mimeType: string;
    sizeInBytes: number;
    createdAt: string;
    isThumbnail: boolean | null;
}

export interface CreateResourceRequest {
    fileName: string;
    mimeType: string;
    sizeInBytes: number;
    isThumbnail?: boolean | null;
}

export interface UploadUrlResponse {
    uid: string;
    uploadUrl: string;
}
